tests/python/contrib/test_hexagon/topi/test_qadd_qsub_qmul.py

```diff
@@ -0,0 +1,246 @@
+    for i in range(b):
+        for j in range(ch):
+            for l in range(cw):
+                for k in range(cc):
+                    if j > 0:
+                        val = l + j + 1.0
+                    else:
+                        val = l
+                    a[i][j][l][k] = val
+
+    return a
+
+
+@tvm.testing.fixture
+def input_np_B(input_shape_B):
+    a = np.zeros(input_shape_B, "float32")
+    shape = input_shape_B
+    b = shape[0]
+    ch = shape[1]
+    cw = shape[2]
+    cc = shape[3]
+    val = 0.0
+    for i in range(b):
+        for j in range(ch):
+            for l in range(cw):
+                for k in range(cc):
+                    if j > 0:
+                        val = l - j - 1.01
+                    else:
+                        val = l
+                    a[i][j][l][k] = val
+    return a
+
+
+@tvm.testing.fixture
+def transformed_input_np_A(input_np_A, input_A_layout):
+    global zero_point_A_val, scale_A_val, input_np_A_quantized
+    zero_point_A_val, scale_A_val, input_np_A_quantized = quantize(input_np_A)
+    return transform_numpy(input_np_A_quantized, "nhwc", input_A_layout)
+
+
+@tvm.testing.fixture
+def transformed_input_np_B(input_np_B, input_B_layout):
+    global zero_point_B_val, scale_B_val, input_np_B_quantized
+    zero_point_B_val, scale_B_val, input_np_B_quantized = quantize(input_np_B)
+    return transform_numpy(input_np_B_quantized, "nhwc", input_B_layout)
+
+
+@tvm.testing.fixture
+def transformed_expected_output_np(expected_output_np, output_layout):
+    global zero_point_M_val, scale_M_val, out_ref_quantized
+    zero_point_M_val, scale_M_val, out_ref_quantized = quantize(expected_output_np)
+    return transform_numpy(out_ref_quantized, "nhwc", output_layout)
+
+
+def hexagon_wrapper_allocation(
+    device, layout, axis_separators, tensor_shape=None, data=None, transformed_data=None, dtype=None
+):
+    """Input layout can either be nhwc-8h8w32c-2d or nhwc"""
+    if layout == "nhwc-8h8w32c-2d":
+        data_nd = allocate_hexagon_array(
+            device,
+            tensor_shape=tensor_shape,
+            data=transformed_data,
+            dtype=dtype,
+            axis_separators=axis_separators,
+            mem_scope="global.vtcm",
+        )
+    elif layout == "nhwc":
+        data_nd = allocate_hexagon_array(
+            device,
+            data=data,
+        )
+    return data_nd
+
+
+class TestAddSubtractMultiplyBroadcast2d:
+    (
+        input_shape_A,
+        input_shape_B,
+        input_A_layout,
+        input_B_layout,
+        output_layout,
+        dtype,
+    ) = tvm.testing.parameters(
+        # no broadcast needed - short input
+        # broadcast axis 2 in one input
+        (
+            [1, 8, 8, 32],
+            [1, 8, 8, 1],
+            "nhwc-8h8w32c-2d",
+            "nhwc",
+            "nhwc-8h8w32c-2d",
+            "uint8",
+        ),
+    )
+
+    op_name = tvm.testing.parameter("add")  # , "subtract", "multiply")

```

Comment:
For this specific input values, the error only occurs with add. subtract or multiply don't reproduce the same error